src/app/redux/common/actions.ts

```diff
@@ -0,0 +1,22 @@
+import { createAction } from '@reduxjs/toolkit';
+import { IMessage, SEND_MESSAGE, DELETE_MESSAGE, IChatActionTypes } from './types';
+
+// TypeScript infers that this function is returning SendMessageAction
```

Comment:
I guess this is fine, maybe smart? Definitely weird. Whatever the redux team recommends, I think is probably good.